### DISCUSSION TITLE
[plugin] plugins should not add system wide aliases

### DISCUSSION THREAD

**Discussion by tobocop2** [ohmyzsh/ohmyzsh#12324] [Help] [answered]
I installed the yarn plugin and was freaked out after some of my common commands became aliases to yarn commands. This is very intrusive and alarmed me.  I thought my system got compromised before i verified where the aliases came from. 

Do folks agree that system wide aliases should not be managed by any zsh plugins? These will take precedence over executables with the same name and that's not ideal. For example `yt` for me is the executable path for `yewtube`  and installing the yarn plugin made it `yarn test` and i just `unaliased`'d it for now. 

Aliases should be communicated better to the user and should be something that can be easily opted out of.

**Comment by rwmitchell** [comment]
plugins/yarn/README.md

seems to document each of the aliases pretty well.  You didn't read the README?

**Comment by carlosala** [comment, marked as answer]
If you want to skip any aliases (per plugin, or in general all aliases defined by `oh-my-zsh.sh`) you can read about it here! => https://github.com/ohmyzsh/ohmyzsh?tab=readme-ov-file#skip-aliases

**Reply by tobocop2** [reply]
thanks i didn't read the docs.  I've been using this framework for years so it's just not something I ever noticed before until recently 😬  so I didn't think it was necessary to consult the docs when installing a plugin. In fact I had no idea plugins by default could even add global aliases to the system and am now going to be mindful of it. I think that's not ideal as a default.

**Reply by carlosala** [reply]
It's indeed one of the primary purposes of plugins! Even though, we realized that many people were only interested on other features (like completion, autoloading of tools, etc) and we added the option to skip aliases.
Thanks!

**Reply by mcornella** [reply]
Most of the appeal of big popular plugins is the system of aliases they define, such as the git plugin, which allow for standardized plugins across many different setups, for example if your colleagues also  use Oh My Zsh, or you deploy those on servers or whatever.

We know that's not for everybody, which is why we added the option of turning off aliases (with zstyle, which allows for quite a big flexibility).

Every plugin has its own README file, as you'll see, and yarn specifically doesn't have much appeal other than aliases.

**Reply by tobocop2** [reply]
i appreciate you folks for being so quick to respond. Thanks again. I just enabled the yarn plugin thinking it would give me auto completion is all so the aliases came by surprise since I've never consulted the docs outside of just installing the framework 🤣
